source/opt/folding_rules.cpp

```diff
@@ -2507,24 +2507,16 @@ FoldingRule RedundantFMix() {
   };
 }
 
-// This rule handles addition of zero for integers.
-FoldingRule RedundantIAdd() {
-  return [](IRContext* context, Instruction* inst,
+FoldingRule RedundantBinaryLRhs0(spv::Op op, size_t arg) {
```

Comment:
The name is hard to parse because of the acronyms. Also, a function level comment is expected useful.

```suggestion
// Returns a folding rule that folds the instruction to the operand not being checked if the operand that is checked is zero.
FoldingRule RedundantBinaryOpWithZeroOperand(spv::Op op, bool checkFirstOperand) {
```